src/libs/ReportSecondaryActionUtils.ts

```diff
@@ -0,0 +1,330 @@
+    isProcessingReport,
+    isReportApproved,
+    isSettled,
+} from './ReportUtils';
+import {getSession} from './SessionUtils';
+import {allHavePendingRTERViolation, isDuplicate, isOnHold as isOnHoldTransactionUtils, shouldShowBrokenConnectionViolationForMultipleTransactions} from './TransactionUtils';
+
+function isSubmitAction(report: Report, policy: Policy): boolean {
+    const isExpense = isExpenseReport(report);
+
+    if (!isExpense) {
+        return false;
+    }
+
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+
+    if (!isSubmitter && !isApprover) {
+        return false;
+    }
+
+    const isOpen = isOpenReport(report);
+
+    if (!isOpen) {
+        return false;
+    }
+
+    const autoReportingFrequency = getCorrectedAutoReportingFrequency(policy);
+
+    const isScheduledSubmitEnabled = policy?.harvesting?.enabled && autoReportingFrequency !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL;
+
+    return !!isScheduledSubmitEnabled;
+}
+
+function isApproveAction(report: Report, policy: Policy, reportTransactions: Transaction[], violations: OnyxCollection<TransactionViolation[]>): boolean {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isProcessing = isProcessingReport(report);
+    const hasDuplicates = reportTransactions.some((transaction) => isDuplicate(transaction.transactionID));
+
+    if (isExpense && isApprover && isProcessing && hasDuplicates) {
+        return true;
+    }
+
+    const transactionIDs = reportTransactions.map((t) => t.transactionID);
+
+    const hasAllPendingRTERViolations = allHavePendingRTERViolation(transactionIDs, violations);
+
+    if (hasAllPendingRTERViolations) {
+        return true;
+    }
+
+    const isAdmin = policy?.role === CONST.POLICY.ROLE.ADMIN;
+
+    const shouldShowBrokenConnectionViolation = shouldShowBrokenConnectionViolationForMultipleTransactions(transactionIDs, report, policy, violations);
+
+    const userControllsReport = isApprover || isAdmin;
+    return userControllsReport && shouldShowBrokenConnectionViolation;
+}
+
+function isUnapproveAction(report: Report, policy: Policy): boolean {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isApproved = isReportApproved({report});
+
+    return isExpense && isApprover && isApproved;
+}
+
+function isCancelPaymentAction(report: Report): boolean {
+    const isExpense = isExpenseReport(report);
+
+    if (!isExpense) {
+        return false;
+    }
+
+    const isPaidElsewhere = report.stateNum === CONST.REPORT.STATE_NUM.APPROVED && report.statusNum === CONST.REPORT.STATUS_NUM.REIMBURSED;
+
+    if (isPaidElsewhere) {
+        return true;
+    }
+
+    const isPaymentProcessing = true; // TODO
+    const hasDailyNachaCutoffPassed = false; // TODO
```

Comment:
@luacmartins I'm not sure how to determine this value

Reply:
The cutoff is 23:45 PM UTC

Reply:
Yes, you can probably use [getIOUActionForReportID](https://github.com/Expensify/App/blob/c406beed64b4a5f4b7959d0b63b163d9d0dea5fc/src/libs/ReportActionsUtils.ts#L1631) to get the IOU action and check if it's a [pay action](https://github.com/Expensify/App/blob/c406beed64b4a5f4b7959d0b63b163d9d0dea5fc/src/libs/ReportActionsUtils.ts#L1072) and then use the reportAction date